libgalois/include/katana/LC_CSR_CSC_Labeled_Graph.h

```diff
@@ -337,34 +337,47 @@ class LC_CSR_CSC_Labeled_Graph
       GraphNode value =
           in_edges ? this->getInEdgeDst(mid) : this->getEdgeDst(mid);
       if (value == key) {
-        // Update bounds for last only if we're searching for the first element.
         if (finding_first) {
+          // Update bounds for last only if we're searching for the first
```

Comment:
the STL uses lower_bound and upper_bound to achieve the same idea. It might be more maintainable to use those (or analogues thereof) instead of merging finding first/any/last into one function.

Reply:
I agree. The complexity and corner cases are getting out of hand here. Better to rewrite it using STL's upper_bound and lower_bound.

Reply:
That's a good idea. I don't see any benefits of maintaining our own code for this. Let's use lower_bound and upper_bound directly.